### PR TITLE
To remove, we must first add

### PR DESCRIPTION
**Pull request by dougbu**
- part of #93
- copy `WebEncoders` into Common repo
- add hacks to work around dotnet/cli#3831 and NuGet/Home#3118

### CODE REVIEW

**Comment by anurse** [comment]
:shipit:

File: makefile.shade [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+              var fileName = Path.GetFileName(file);
+              var targetFile = Path.Combine(targetDirectory, fileName);
+              File.Copy(file, targetFile, overwrite: false);
+          }
+      }
+
+      // Update the .nuspec file.
+      var nuspecFile = Files.Include(Path.Combine(extractToDirectory, "*.nuspec")).First();
+      var nuspec = new List<string>(File.ReadAllLines(nuspecFile));
+      var endOfMetadata = nuspec.FindIndex(l => l.Contains("</metadata>"));
```
Comment (pranavkm) [inline makefile.shade]: Does using `XDocument` work here? Would make it more resilient to the format of the underlying xml.

**Comment by pranavkm** [comment]
:shipit:

File: makefile.shade (hunk 1 above)
Comment (dougbu) [inline makefile.shade]: I'll give it a try...

**Comment by dougbu** [comment]
FYI :up::date:

Switched to `XDocument`. Since generated content is the same and I've got sign-offs, will push immediately.

**Comment by dougbu** [comment]
fe2bf2c

File: makefile.shade [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+      var metadata = package.Element(nuspecNamespace + "metadata");
+      var files = new XElement(
+          nuspecNamespace + "files",
+          new XAttribute("include", "**/*.resx"),
+          new XAttribute("buildAction", "EmbeddedResource"));
+      var contentFiles = new XElement(nuspecNamespace + "contentFiles", files);
+
+      metadata.Add(contentFiles);
+      nuspec.Save(nuspecFile);
```
Comment (pranavkm) [inline makefile.shade]: :100:

**Pull request closed without merging**